rename kinvolk => flatcar org, remove submodules references

t-lo opened the pull request:
# Use `flatcar` instead of `kinvolk` github org, remove references to git submodules in `scripts` repo

This change removes any "submodules" and "gitmodules" reference to `coreos-overlay` and `portage-stable` ebuild repositories as these are now sub-directories in the `scripts` repo.

Further, all references and links to the `kinvolk` github org and `kinvolk.io` website are replaced by `flatcar` where applicable. For all references to `ignition`, upstream coreos ignition is now referenced; references to `ct` are replaced by references to upstream coreos butane.

krnowak commented on docs/provisioning/ignition/boot-process.md:
```diff
@@ -111,9 +111,9 @@ Here is an example config with an additional `links` entry that ensures that the
+[gptprio.next]: https://github.com/flatcar/scripts/blob/9e1c23f3f44d2751076e770f43f7a6db05d49652/build_library/grub.cfg#L132
 [grub]: https://www.gnu.org/software/grub/
-[grub-config]: https://github.com/kinvolk/scripts/blob/9e1c23f3f44d2751076e770f43f7a6db05d49652/build_library/grub.cfg
+[grub-config]: https://github.com/flatcar/scripts/blob/9e1c23f3f44d2751076e770f43f7a6db05d49652/build_library/grub.cfg
```
```suggestion
[grub-config]: https://github.com/flatcar/scripts/blob/80e49d190ff99e8c489bbf420dc2bc248ae553e3/build_library/grub.cfg
```

krnowak commented on docs/reference/developer-guides/sdk-modifying-flatcar.md:
```diff
@@ -680,8 +674,8 @@ Take a look at the [SDK bootstrap process](sdk-bootstrapping) to learn how to bu
 
-[coreos]: https://github.com/flatcar/coreos-overlay
-[portage]: https://github.com/flatcar/portage-stable
+[coreos]: https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay
```
```suggestion
[coreos]: https://github.com/flatcar/scripts/tree/main/sdk_container/src/third_party/coreos-overlay
```

krnowak commented on docs/reference/developer-guides/sdk-modifying-flatcar.md:
```diff
@@ -680,8 +674,8 @@ Take a look at the [SDK bootstrap process](sdk-bootstrapping) to learn how to bu
-[coreos]: https://github.com/flatcar/coreos-overlay
-[portage]: https://github.com/flatcar/portage-stable
+[coreos]: https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay
+[portage]: https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/portage-stable
```
```suggestion
[portage]: https://github.com/flatcar/scripts/tree/main/sdk_container/src/third_party/portage-stable
```

krnowak commented on docs/reference/supply-chain.md:
```diff
@@ -121,12 +121,12 @@ Renewing the image signing key requires split secrets of multiple maintainers.
-   2. Package definition (ebuild) repositories "pinned" as git submodules to a commit in the build automation repo.
-      Package definitions include the [coreos-overlay](https://github.com/flatcar-linux/coreos-overlay) and [portage-stable](https://github.com/flatcar-linux/portage-stable) repositories.
+      It includes all package definitions (ebuilds) in subdirectories.
+      Package definitions include the [portage-stable](https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/portage-stable) and [coreos-overlay](https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay) ebuild "repositories".
```
```suggestion
      Package definitions include the [portage-stable](https://github.com/flatcar/scripts/tree/main/sdk_container/src/third_party/portage-stable) and [coreos-overlay](https://github.com/flatcar/scripts/tree/main/sdk_container/src/third_party/coreos-overlay) ebuild "repositories".
```

t-lo commented on docs/provisioning/cl-config/_index.md:
```diff
@@ -58,9 +58,9 @@ As shown in this diagram, `ct` is manually invoked only when users are manually
 
-The Container Linux Config Transpiler abstracts the details of configuring Flatcar Container Linux. It's responsible for transforming a Container Linux Config written by a user into an Ignition Config to be consumed by instances of Flatcar Container Linux.
+The Butane Config Transpiler abstracts the details of configuring Flatcar Container Linux. It's responsible for transforming a Container Linux Config written by a user into an Ignition Config to be consumed by instances of Flatcar Container Linux.
 
-The Container Linux Config Transpiler command line interface, `ct` for short, can be downloaded from its [GitHub Releases page][download-ct] or used via Docker (`cat example.yaml | docker run --rm -i ghcr.io/flatcar/ct:latest --platform=YOURPLATFORM`).
+The Butane Config Transpiler command line interface, `butane` for short, can be downloaded from its [GitHub Releases page][download-butane] or used via Docker (`cat example.yaml | docker run --rm -i quay.io/coreos/butane:release --platform=YOURPLATFORM`).
```
I think we should update this as good as we can - there's another PR going much deeper here: https://github.com/flatcar/flatcar-docs/pull/299 - and fix the fallout afterwards. Otherwise it will always be too big to change :)

krnowak replied:
I mean, we still support both Container Linux Config Transpiler and Butane Transpiler. And I think this document is about the former, so it should not be talking about `butane` (other than being a recommended replacement of `ct`).

t-lo replied:
Hmmm, that's a good point. I think find-and-replace got the better of me here.

I believe we should move the `ct` section below the `butane` section and call out that people should be using `butane`  since `ct` is outdated. But that's definitely stuff for the other PR, #299 .

t-lo commented on docs/provisioning/ignition/_index.md:
```diff
@@ -32,8 +32,6 @@ Because Ignition only runs once, there's no reason for it to incorporate dynamic
 
 Instead, use Ignition to write static files and leverage systemd's environment variable expansion to insert dynamic data. The Ignition config should install a service which fetches the necessary runtime data, then any services which need this data (such as etcd or fleet) can rely on the installed service and source in their output. The result is that the data is only collected if and when it is n ...
 
-The lack of variable substitution in Ignition has an added benefit of leveling the playing field when it comes to compute providers. The user's experience is no longer crippled because the metadata for their platform isn't supported. It is possible to write a [custom metadata agent][custom-agent] to fetch the necessary data.
```
Correct.

t-lo commented on docs/provisioning/ignition/boot-process.md:
```diff
@@ -111,9 +111,9 @@ Here is an example config with an additional `links` entry that ensures that the
+[grub-config]: https://github.com/flatcar/scripts/blob/9e1c23f3f44d2751076e770f43f7a6db05d49652/build_library/grub.cfg
 [linux-startup]: https://en.wikipedia.org/wiki/Linux_startup_process
-[search-guid]: https://github.com/kinvolk/scripts/blob/9e1c23f3f44d2751076e770f43f7a6db05d49652/build_library/grub.cfg#L73-L78
+[search-guid]: https://github.com/flatcar/scripts/blob/9e1c23f3f44d2751076e770f43f7a6db05d49652/build_library/grub.cfg#L73-L78
```
Absolutely. This PR only fixes the repo names / github orgs, but there definitely should be follow-up work.

t-lo commented on docs/setup/clusters/architectures.md:
```diff
@@ -59,7 +59,7 @@ systemd:
 This file is used to provision your local Flatcar Container Linux machine on its first boot. This sets up and enables the Docker API, which is how you can use Docker on your laptop. The Docker CLI manages containers running within the VM, *not* on your personal operating system.
 
-Using the CL Config Transpiler, or `ct`, ([download][ct-download]) convert the above yaml into an [Ignition][ignition-getting-started]. Alternatively, copy the contents of the Igntion tab in the above example. Once you have the Ignition configuration file, pass it to your provider ([complete list of supported Ignition platforms][ignition-supported]).
+Using the Butane Config Transpiler, or `butane`, ([download][butane-download]) convert the above yaml into an [Ignition][ignition-getting-started]. Alternatively, copy the contents of the Igntion tab in the above example. Once you have the Ignition configuration file, pass it to your provider ([complete list of supported Ignition platforms][ignition-supported]).
```
https://github.com/flatcar/flatcar-docs/pull/299 takes care of that, let's rebase that one and merge afterwards. There are a number of comments open in https://github.com/flatcar/flatcar-docs/pull/299; I'll have a look after this one is merged.

krnowak replied:
Sounds good.

t-lo commented:
Thank you @krnowak for the thorough review and for your awesome suggestions!
Re: https://github.com/flatcar/flatcar-docs/pull/299, I only became aware of that PR after filing this one. I'll take care re-basing and of addressing the comments in that PR after this one is merged.

t-lo commented on docs/installing/community-platforms/notes-for-distributors.md:
```diff
@@ -42,25 +42,25 @@ There are two predominant ways that a Flatcar Container Linux image can be easil
-[ign-platforms]: https://github.com/kinvolk/ignition/blob/master/doc/supported-platforms.md
-[coreos-metadata]: https://github.com/kinvolk/afterburn/
-[ct]: https://github.com/kinvolk/container-linux-config-transpiler
+[ign-platforms]: https://github.com/coreos/ignition/blob/main/docs/supported-platforms.md
```
Added.

krnowak commented on docs/installing/community-platforms/notes-for-distributors.md:
```diff
@@ -38,29 +38,29 @@ There are two predominant ways that a Flatcar Container Linux image can be easil
 ### Ignition
 
-[Ignition][ignition] is a tool that acquires a JSON config file when a machine first boots, and uses this config to perform tasks such as formatting disks, creating files, modifying and creating users, and adding systemd units. How Ignition acquires this config file varies per-platform, and it is highly recommended that providers ensure Ignition has [support for their platform][ign-platforms].
+[Ignition][ignition] is a tool that acquires a JSON config file when a machine first boots, and uses this config to perform tasks such as formatting disks, creating files, modifying and creating users, and adding systemd units. How Ignition acquires this config file varies per-platform, and it is highly recommended that providers ensure Ignition supports their respective platform. In addition to  ...
```
```suggestion
[Ignition][ignition] is a tool that acquires a JSON config file when a machine first boots, and uses this config to perform tasks such as formatting disks, creating files, modifying and creating users, and adding systemd units. How Ignition acquires this config file varies per-platform, and it is highly recommended that providers ensure Ignition supports their respective platform. In addition to providers supported by [upstream Ignition][ign-platforms], Flatcar [supports](https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0018-revert-internal-oem-drop-noop-OEMs.patch) cloudsigma, hyperv, interoute, niftycloud, rackspace[-onmetal], and vagrant.
```

krnowak commented on docs/provisioning/ignition/_index.md:
```diff
@@ -46,15 +44,14 @@ Ignition can read its config from a number of different locations, but only from
 ## Where is Ignition supported?
 
-The [full list of supported platforms][supported-platforms] is provided and will be kept up-to-date as development progresses.
+In addition to providers supported by [upstream Ignition][ignition-supported], Flatcar [supports](https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0018-revert-internal-oem-drop-noop-OEMs.patch) cloudsigma, hyperv, ineroute, niftycloud, rackspace[-onmetal], and vagrant.
```
```suggestion
In addition to providers supported by [upstream Ignition][ignition-supported], Flatcar [supports](https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0018-revert-internal-oem-drop-noop-OEMs.patch) cloudsigma, hyperv, interoute, niftycloud, rackspace[-onmetal], and vagrant.
```

krnowak commented on docs/setup/clusters/architectures.md:
```diff
@@ -59,7 +59,8 @@ systemd:
 
-Using the CL Config Transpiler, or `ct`, ([download][ct-download]) convert the above yaml into an [Ignition][ignition-getting-started]. Alternatively, copy the contents of the Igntion tab in the above example. Once you have the Ignition configuration file, pass it to your provider ([complete list of supported Ignition platforms][ignition-supported]).
+Using the Butane Config Transpiler, or `butane`, ([download][butane-download]) convert the above yaml into an [Ignition][ignition-getting-started]. Alternatively, copy the contents of the Igntion tab in the above example. Once you have the Ignition configuration file, pass it to your provider.
+In addition to providers supported by [upstream Ignition][ignition-supported], Flatcar [supports](https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0018-revert-internal-oem-drop-noop-OEMs.patch) cloudsigma, hyperv, ineroute, niftycloud, rackspace[-onmetal], and vagrant.
```
```suggestion
In addition to providers supported by [upstream Ignition][ignition-supported], Flatcar [supports](https://github.com/flatcar/scripts/blob/main/sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0018-revert-internal-oem-drop-noop-OEMs.patch) cloudsigma, hyperv, interoute, niftycloud, rackspace[-onmetal], and vagrant.
```
